### PR TITLE
TTL accepts non-negative integers including ZERO

### LINKED ISSUE #491: Zero TTL Not Working
According to the spec [here](https://tools.ietf.org/html/draft-ietf-webpush-protocol-12#section-5.2), a zero TTL means that the message is not stored by FCM and only delivered to endpoints that are online at that moment.

In my testing using the node web-push library, when I am setting TTL to ZERO in the options object as follows :
`const options = {
        gcmAPIKey: <api_key>,
        vapidDetails: {
            subject: 'mailto:example@web-push-node.org',
            publicKey: <public_key>,
            privateKey: <private_key>
        },
        TTL: 0
}`

But receiving the web-push even after keeping the device offline for half hour.

Tested for both latest Chrome and Firefox.

Please help!

### PR DESCRIPTION
**Pull request by renegadeHomie**
Fixes #491 
#284 

The if condition to check if options.TTL is present doesn't honour ZERO. This commit makes TTL accepts non-negative integers including ZERO.

### CODE REVIEW

File: src/web-push-lib.js [hunk 1]
```diff
@@ -155,7 +155,7 @@ WebPushLib.prototype.generateRequestDetails = function(subscription, payload, op
         currentVapidDetails = options.vapidDetails;
       }
 
-      if (options.TTL) {
+      if (Number.isInteger(options.TTL) && options.TTL > -1) {
```
Comment (marco-c) [inline src/web-push-lib.js]:
```suggestion
      if (options.TTL !== undefined) {
```

And then:
```
timeToLive = Number(options.TTL)
if (timeToLive < 0) {
  throw new Error("TTL should be a number and should be at least 0");
}
```

Comment (renegadeHomie) [reply src/web-push-lib.js]: Sure, will add the suggested changes.

Before going ahead, I want to point out the implications this change will have on the Impression stats for people using the library.

So, currently whoever is sending TTL=0, the messages are actually getting delivered for a timespan for 4 weeks as that is the default.

**With this change, the stats being recorded by such end-users for their bulk campaigns sent using this library will drastically drop.**

Would want to hear your thoughts on the same.

File: src/web-push-lib.js [hunk 2]
```diff
@@ -190,6 +193,7 @@ WebPushLib.prototype.generateRequestDetails = function(subscription, payload, op
       }
     }
 
+
```
Comment (marco-c) [inline src/web-push-lib.js]: Nit: Please remove this unrelated newline

File: src/web-push-lib.js (hunk 1 above)
Comment (marco-c) [reply src/web-push-lib.js]: I think it's an acceptable drawback, we need to honor the choice of the users.

Comment (renegadeHomie) [reply src/web-push-lib.js]: Agree, this aligns the library with the VAPID [spec](https://tools.ietf.org/html/draft-ietf-webpush-protocol-12#section-5.2)

**Comment by renegadeHomie** [comment]
Is there anything else I should add i.e. update the README or something?

File: src/web-push-lib.js [hunk 3]
```diff
@@ -155,8 +155,11 @@ WebPushLib.prototype.generateRequestDetails = function(subscription, payload, op
         currentVapidDetails = options.vapidDetails;
       }
 
-      if (options.TTL) {
-        timeToLive = options.TTL;
+      if (options.TTL !== undefined) {
+        timeToLive = Number(options.TTL);
+        if (timeToLive < 0) {
+          throw new Error("TTL should be a number and should be at least 0");
```
Comment (marco-c) [inline src/web-push-lib.js]: Just one needed change, for our eslint rules these should be single quotes.

```suggestion
          throw new Error('TTL should be a number and should be at least 0');
```

**Comment by renegadeHomie** [comment]
@marco-c Says branch is outdated. It looks good otherwise. Please merge the PR.

**Pull request merged**